client/landing/stepper/declarative-flow/domain-upsell.ts

```diff
@@ -56,9 +58,12 @@ const domainUpsell: Flow = {
 				case 'domains':
 					if ( providedDependencies?.deferDomainSelection ) {
 						try {
-							await updateLaunchpadSettings( siteSlug, {
-								checklist_statuses: { domain_upsell_deferred: true },
-							} );
+							const siteIdentifier = siteSlug || siteId;
+							if ( siteIdentifier ) {
+								await updateLaunchpadSettings( siteIdentifier, {
+									checklist_statuses: { domain_upsell_deferred: true },
+								} );
+							}
```

Comment:
This may fail silently if both params are missing, but that is not a regression.